lib/fizzy/execute.cpp

```diff
@@ -378,6 +378,55 @@ execution_result execute(Instance& instance, FuncIdx func_idx, std::vector<uint6
             labels.push_back(label);
             break;
         }
+        case Instr::if_:
+        {
+            if (static_cast<uint32_t>(stack.pop() != 0))
+            {
+                const auto arity = read<uint8_t>(immediates);
+                const auto target_pc = read<uint32_t>(immediates);
+                const auto target_imm = read<uint32_t>(immediates);
+                read<uint32_t>(immediates);  // Omit the immediates for else instruction.
+                read<uint32_t>(immediates);  // Omit the immediates for else instruction.
+                LabelContext label{code.instructions.data() + target_pc,
+                    code.immediates.data() + target_imm, arity, stack.size()};
+                labels.emplace_back(label);
+            }
+            else
+            {
+                const auto arity = read<uint8_t>(immediates);
+                const auto target_pc = read<uint32_t>(immediates);
+                const auto target_imm = read<uint32_t>(immediates);
+                const auto target_else_pc = read<uint32_t>(immediates);
+                const auto target_else_imm = read<uint32_t>(immediates);
+
+                if (target_else_pc != 0 && target_else_imm != 0)  // If else block defined.
```

Comment:
would be enough to check just one?

Reply:
One should be enough. This assumes `else` cannot be first instruction.